src/Generator.fs

```diff
@@ -148,11 +221,27 @@ module Page =
 
             do! IO.writeFile dist content
 
+            let layout = discriminateLayout source
+
+            let chooseDate (fm: Parser.FrontMatter option) alt =
+                match fm with
+                | Some fm ->
+                    match fm.date with
+                    | None -> alt
+                    | Some x -> x
+                | None -> alt
```

Comment:
I want to refactor this to use pattern matching function.
throughout, not just here.

Reply:
Another time